### ISSUE TITLE
Code snippet to support VeraSecure built in siren

### ISSUE THREAD

**Issue by patrik3k** [pavoni/pyvera#84]
Hi,
first off I'm a total noob on github so this is probably the wrong way to handle this. Secondly, big thanks for writing this library.

Now to the topic. 
I have a VeraSecure controller that is connected to Home Assistant that is using your library. The VeraSecure have an built in siren that is triggered with the different modes. It can also be turned on and off via Luup calls. BUT the siren is now added as a generic device that can be armed. 

The siren is not armed in that sense in Vera but tightly connected to the modes. 

To automate siren and the usage of the siren the siren should be treated as a switch.

In __init__.py 
When initiating all devices the siren is not recognised by the right category and is initilized as a VeraDevice. 

My suggestion is to add  the correct category  on row 32

CATEGORY_POWER_METER = 21
CATEGORY_VERA_SIREN = 24
CATEGORY_UV_SENSOR = 28

And add the correct device class for it.  Rows 160-170isch 

                elif device_category == CATEGORY_SCENE_CONTROLLER:
                    device = VeraSceneController(item, self)
                elif device_category == CATEGORY_VERA_SIREN:
                    device = VeraSwitch(item, self)
                else:
                    device = VeraDevice(item, self)
                self.devices.append(device)
                if device_category != CATEGORY_SWITCH and device_category != CATEGORY_VERA_SIREN and device.is_armable:
                    self.devices.append(VeraArmableDevice(item, self))
            else:
                self.devices.append(VeraDevice(item, self))

Im runnig this myself and I can now controll and detect when the siren is invoked with no other changes.

Let me know if this is interesting and if you need more information.

Thanks for great job on this,
Patrik Ekström

**Comment by pavoni** [comment]
Cool - thanks for contributing! 

The `correct` way to do this is for you to fork the library on github to get your own version. Create a new branch on your version - commit your changes - and then send me a pull request with the code above.

The advantage of that is when I merge the code into my repo - I know I'm using exactly the code you've tested - pllus you get added as a contributor and get credit for your work!

If that's too much hassle - I'm fine to create a branch with your changes that you can test.

BTW I'd probably change the code very slightly (haven't tested this) to follow the pattern of the sensor category test.
```
                if device_category == CATEGORY_DIMMER:
                    device = VeraDimmer(item, self)
                elif ( device_category == CATEGORY_SWITCH or
                       device_category == CATEGORY_VERA_SIREN):
                    device = VeraSwitch(item, self)
                elif device_category == CATEGORY_THERMOSTAT:
                    device = VeraThermostat(item, self)
                elif device_category == CATEGORY_LOCK:
                    device = VeraLock(item, self)
                elif device_category == CATEGORY_CURTAIN:
                    device = VeraCurtain(item, self)
                elif device_category == CATEGORY_ARMABLE:
                    device = VeraBinarySensor(item, self)
                elif (device_category == CATEGORY_SENSOR or
                      device_category == CATEGORY_HUMIDITY_SENSOR or
                      device_category == CATEGORY_TEMPERATURE_SENSOR or
                      device_category == CATEGORY_LIGHT_SENSOR or
                      device_category == CATEGORY_POWER_METER or
                      device_category == CATEGORY_UV_SENSOR):
                    device = VeraSensor(item, self)
                elif device_category == CATEGORY_SCENE_CONTROLLER:
                    device = VeraSceneController(item, self)
                else:
                    device = VeraDevice(item, self)
                self.devices.append(device)
                if (device.is_armable and not (
                    device_category == CATEGORY_SWITCH or 
                    device_category == CATEGORY_VERA_SIREN):
                    self.devices.append(VeraArmableDevice(item, self))
            else:
                self.devices.append(VeraDevice(item, self))
```

Let me know what you'd like to do...

**Comment by patrik3k** [comment]
Hi Greg,
Thanks for quick and very informative response. Its a long time since my programing days and I heard CVS is out of style nowadays:)

But its great, I learned something new. I have created a new branch and a pull request for you. I really hope I did it correctly. Your snippet looked much cleaner and consistent so I used it, there was just a  missing closing parenthasis  in the if statement that I fixed.  

I have only tested using your library from within home assistant platform. Changing the behaviour for the siren in your library was the only solution I found to make it work. I hope this doesn’t ruin it in any other way. 

Best regards
Patrik Ekström
CEO
Top of Minds
Sturegatan 4
114 35 Stockholm, Sweden
mobile: +46 708 406952
